pymc_marketing/mmm/fourier.py

```diff
@@ -0,0 +1,653 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Fourier seasonality transformations.
+
+This modules provides Fourier seasonality transformations for use in
+Marketing Mix Models. The Fourier seasonality is a set of sine and cosine
+functions that can be used to model periodic patterns in the data.
+
+There are two types of Fourier seasonality transformations available:
+
+- Yearly Fourier: A yearly seasonality with a period of 365 days
+- Monthly Fourier: A monthly seasonality with a period of 30 days
```

Comment:
What about using https://pandas.pydata.org/docs/reference/api/pandas.Period.days_in_month.html ? Sometimes long and short months can mess up the estimation (it is less worrying for years)

Reply:
Yes let s keep it like this